### ISSUE TITLE
[Feature Request] Proxy Settings

### ISSUE THREAD

**Issue by lgjint** [Nheko-Reborn/nheko#1808]
### The Problem

Currently, nheko does not provide an option for users to configure proxy settings. The proxy settings are very useful for users like me in areas with restricted internet, to be able to bypass censorship and use the Matrix.

### The Solution

I hope that nheko can provide the option of manually setting the proxy or using the system proxy in the settings like Neochat.
![image](https://github.com/user-attachments/assets/12733baa-69db-404f-b5b3-72b4ca0ba550)

### Alternatives

N/A

### Additional context

N/A

### Happens in the latest version

- [X] Yes, this feature is still missing.

**Comment by uis246** [comment]
Nheko uses curl, so you can set proxy with enviroment variable. `all_proxy=socks5h://host:port`

**Comment by voidastro4** [comment]
> Nheko uses curl, so you can set proxy with enviroment variable. `all_proxy=socks5h://host:port`

This results in name resolution failures and nothing works. When properly isolated with no access to anything other than socks5.

If things do work for you with this setting it's even worse because DNS leaks are guaranteed.

**Comment by ffs-akl** [comment]
> Nheko uses curl, so you can set proxy with enviroment variable. `all_proxy=socks5h://host:port`

  Do setting this in ~/.config/nheko/nheko.conf  make the proxy work ?

**Comment by skobkin** [comment]
I double this issue.

Nheko works also on Windows where passing env variables only to specific executable is not very convenient. It's better to support setting that as an option in the GUI like other messengers often do.

**Comment by uis246** [comment]
Can't you pass arbitrary environment variables in desktop shortcut in windows? I'm not very familiar with it.

**Comment by arrowd** [comment]
Solved via https://stackoverflow.com/questions/3036325/can-i-set-an-environment-variable-for-an-application-using-a-shortcut-in-windows#comment92433436_34769146

**Comment by jez9999** [comment]
I agree with the comments that passing an env variable is very user-unfriendly.  I mean come on, how hard is it to just provide a simple UI for this with a few textboxes?

**Comment by ffs-akl** [comment]
sudo pacman -S  proxychains-ng
sudo nano /etc/proxychains.conf
add the  `socks5  127.0.0.1 xxx` (you should use your porxy port instead of xxx) to  the end of /etc/proxychains.conf
lanuch nheko with command `proxychains4  nheko`
Note:  I am using arch.    you should refer to your linux system

**Comment by skobkin** [comment]
This request is about UI/UX.

Proxychains, ENV vars or any other way to force app to use proxy are irrelevant here.

UI/UX helps to attract users to the app.
Workarounds help one person to solve their specific problem and do not work for anyone else unless they actively look for them and ready to use them instead of just switching the app.

If someone creates an issue like this, they don't want to solve the problem for themselves. They want the app to be better for everyone. Otherwise they'd go to ChatGPT or StackOverflow.
